### DISCUSSION TITLE
Implement VoiceVLAN

### DISCUSSION THREAD

**Discussion by JeanKean-art** [netbox-community/netbox#18223] [Ideas]
Could you create a new specific type of vlan name VoiceVLAN on a Device interface ?
BR

**Comment by candlerb** [comment]
You can already create a VLAN called "VoiceVLAN".

You can add this to a device interface: just set the 802.1Q mode to Access or Tagged as appropriate, and enable that VLAN on the interface.

Is there any other functionality which you require to differentiate a voice VLAN? If so, you can look at using Tags or Custom Fields.

**Reply by quique** [reply]
I believe the OP refers to a Multi-VLAN Access Port.

This sounds kind of contradictory, but it does exist.
Normally we think that a port carrying traffic for more than one VLAN is by definition a trunk.
However Cisco allows us to have two VLANs connected to one port, if one of those two VLANs is a voice VLAN.

Switch --- Phone --- PC

The PC sends traffic untagged just as it would over an 802.1q native VLAN, while the phone adds a .1Q tag to its frames.
From a configuration perspective on the catalyst switch, technically it’s an access port.

See https://networkengineering.stackexchange.com/questions/65222/whats-the-difference-between-voice-vlan-and-access-vlan for a better explanation.

AFAIK there is not evident way to represent this in Netbox:
The `dcim.Interface` class allows us to save the "native" VLAN for the interface in the `untagged_vlan` field.
However, the tagged voice VLAN carried by this interface cannot be saved in the `tagged_vlans` field, because the interface's `mode`  is not `MODE_TAGGED`, but `MODE_ACCESS`.

As a workaround, https://github.com/netbox-community/netbox/discussions/13714 suggests recording this kind of port as tagged, and creating a custom field to indicate that this port is not an actual trunk port.

**Comment by alehaa** [comment]
Honestly, I do not see a good and easy way to implement this right now without making management more complicated. Using an extra field would require users to set two VLANs per port. However, in our setup, we like to define this rule-based. But what is a good way to implement these rules? Should the voice VLAN be dependent on the location/site, or dependent on the access VLAN being set? Are other conditions important? So for now, we just add it to our Ansible playbooks, but do not set it in NetBox.
